code/modules/reagents/chemistry/reagents/impure_reagents.dm

```diff
@@ -7,27 +7,41 @@
 	name = "Chemical Isomers"
 	description = "Impure chemical isomers made from inoptimal reactions. Causes mild liver damage"
 	//by default, it will stay hidden on splitting, but take the name of the source on inverting. Cannot be fractioned down either if the reagent is somehow isolated.
-	chemical_flags = REAGENT_INVISIBLE | REAGENT_SNEAKYNAME | REAGENT_DONOTSPLIT | REAGENT_CAN_BE_SYNTHESIZED
+	chemical_flags = REAGENT_SNEAKYNAME | REAGENT_DONOTSPLIT | REAGENT_CAN_BE_SYNTHESIZED //impure can be synthed, and is one of the only ways to get almost pure impure
 	ph = 3
-	overdose_threshold = 0 //So that they're shown as a problem (?)
+	impure_chem = null
+	inverse_chem = null
+	failed_chem = null
+	metabolization_rate = 0.1 * REM //default impurity is 0.75, so we get 25% converted. Default metabolisation rate is 0.4, so we're 4 times slower.
+	var/liver_damage = 0.5
 
 /datum/reagent/impurity/on_mob_life(mob/living/carbon/C, delta_time, times_fired)
 	var/obj/item/organ/liver/L = C.getorganslot(ORGAN_SLOT_LIVER)
 	if(!L)//Though, lets be safe
 		C.adjustToxLoss(1 * REM * delta_time, FALSE)//Incase of no liver!
 		return ..()
-	C.adjustOrganLoss(ORGAN_SLOT_LIVER, 0.5 * REM * delta_time)
+	C.adjustOrganLoss(ORGAN_SLOT_LIVER, liver_damage * REM * delta_time)
 	return ..()
 
-/datum/reagent/impurity/toxic
-	name = "Toxic sludge"
-	description = "Toxic chemical isomers made from impure reactions. Causes toxin damage"
+//Basically just so people don't forget to adjust metabolization_rate
+/datum/reagent/inverse
+	name = "Toxic monomers"
+	description = "Inverse reagents are created when a reagent's purity is below it's inverse threshold. The are created either during ingestion - which will then replace their associated reagent, or some can be created during the reaction process."
 	ph = 2
+	chemical_flags = REAGENT_SNEAKYNAME | REAGENT_DONOTSPLIT //Inverse generally cannot be synthed - they're difficult to get
+	//Mostly to be safe - but above flags will take care of this. Also prevents it from showing these on reagent lookups in the ui
+	impure_chem = null
+	inverse_chem = null
+	failed_chem = null
+	///how much this reagent does for tox damage too
+	var/tox_damage = 1
 
-/datum/reagent/impurity/toxic/on_mob_life(mob/living/carbon/C, delta_time, times_fired)
-	C.adjustToxLoss(1 * REM * delta_time, FALSE)
+
+/datum/reagent/inverse/on_mob_life(mob/living/carbon/C, delta_time, times_fired)
+	C.adjustToxLoss(tox_damage * REM * delta_time, FALSE)
 	return ..()
```

Comment:
remember that livers won't purge this due to this not being a toxin subtype

Reply:
Maybe I should change that into a bitflag instead.